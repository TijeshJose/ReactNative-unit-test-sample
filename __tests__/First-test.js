import 'react-native';
import React from 'react';
import First from '../src/first';

import renderer from 'react-test-renderer';

//check first page renders correctly
it('First page renders correctly', () => {
    const tree = renderer.create(<First />).toJSON();
    expect(tree).toMatchSnapshot();
});

//check set name function is called properly
it('Check functions are called properly', () => {
    let instance = renderer.create(<First />).getInstance();
    instance.setName("test");
    expect(instance.state.name).toEqual("test");
});