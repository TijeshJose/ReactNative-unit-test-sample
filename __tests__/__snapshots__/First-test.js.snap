// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First page renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    Name
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Enter you name"
    rejectResponderTermination={true}
    testID="name"
    underlineColorAndroid="transparent"
    value=""
  />
  <View>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginTop": 20,
        }
      }
    >
      
    </Text>
  </View>
</View>
`;
